pkg/trait/environment.go

```diff
@@ -99,12 +101,81 @@ func (t *environmentTrait) Apply(e *Environment) error {
 		}
 	}
 
-	if t.Vars != nil {
-		for _, env := range t.Vars {
-			k, v := property.SplitPropertyFileEntry(env)
+	for _, env := range t.Vars {
+		k, v := property.SplitPropertyFileEntry(env)
+		switch {
+		case strings.HasPrefix(v, "configmap:"):
+			path := strings.TrimPrefix(v, "configmap:")
+			setValFromConfigMapKeySelector(&e.EnvVars, k, path)
+		case strings.HasPrefix(v, "secret:"):
+			path := strings.TrimPrefix(v, "secret:")
+			setValFromSecretKeySelector(&e.EnvVars, k, path)
+		default:
 			envvar.SetVal(&e.EnvVars, k, v)
 		}
 	}
-
 	return nil
 }
+
+func setValFromConfigMapKeySelector(vars *[]corev1.EnvVar, envName string, path string) {
```

Comment:
I think we already cover this logic with our internal API: https://github.com/apache/camel-k/blob/ca5239b57c6e450b527204163d6cea33da297f3e/pkg/apis/camel/v1/common_types.go#L315

You may look at the code which is using this struct and reuse the same instead to avoid duplicating the maintenance effort.

Reply:
done